modules/angular-meteor-meteorCollection.js

```diff
@@ -239,48 +248,25 @@ angularMeteorCollections.factory('$meteorCollection', ['AngularMeteorCollection'
       }
 
       var ngCollection = new AngularMeteorCollection(reactiveFunc(), collection);
-      var realOldItems;
 
       function setAutoBind() {
         if (auto) { // Deep watches the model and performs autobind.
           ngCollection.unregisterAutoBind = $rootScope.$watch(function () {
-            if (ngCollection.UPDATING_FROM_SERVER){
-              realOldItems = _.without(ngCollection, 'UPDATING_FROM_SERVER');
+            if (ngCollection.UPDATING_FROM_SERVER) {
               return 'UPDATING_FROM_SERVER';
             }
             return _.without(ngCollection, 'UPDATING_FROM_SERVER');
           }, function (newItems, oldItems) {
-            if (newItems == 'UPDATING_FROM_SERVER')
+            if (newItems == 'UPDATING_FROM_SERVER' ||
```

Comment:
oldItems is 'UPDATING_FROM_SERVER' and new items is a collection is exactly when we set 
self.UPDATING_FROM_SERVER to false  and then apply root scope (see comment above). That means collection should have been saved already  right before in that updateCollection function ([here](https://github.com/Urigo/angular-meteor/commit/7b549c2db91763eaa770282ca20349063833336b#diff-32cf712277b42c3ac88358aa2911a7f4R160))